mono/metadata/sysmath.c

```diff
@@ -316,3 +316,142 @@ ves_icall_System_Math_SplitFractionDouble (gdouble *v)
 {
 	return modf (*v, v);
 }
+
+float
+ves_icall_System_MathF_Acos (float x)
+{
+	return acosf (x);
+}
+
+float
+ves_icall_System_MathF_Acosh (float x)
+{
+	return acoshf (x);
+}
+
+float
+ves_icall_System_MathF_Asin (float x)
+{
+	return asinf (x);
+}
+
+float
+ves_icall_System_MathF_Asinh  (float x)
+{
+	return asinhf (x);
+}
+
+float
+ves_icall_System_MathF_Atan  (float x)
+{
+	return atan (x);
+}
+
+float
+ves_icall_System_MathF_Atan2 (float x, float y)
+{
+	return atan2f (x, y);
+}
+
+float
+ves_icall_System_MathF_Atanh (float x)
+{
+	return atanhf (x);
+}
+
+float
+ves_icall_System_MathF_Cbrt (float x)
+{
+	return cbrtf (x);
+}
+
+float
+ves_icall_System_MathF_Ceiling (float x)
+{
+	return ceilf(x);
+}
+
+float
+ves_icall_System_MathF_Cos (float x)
+{
+	return cosf (x);
+}
+
+float
+ves_icall_System_MathF_Cosh (float x)
+{
+	return coshf (x);
+}
+
+float
+ves_icall_System_MathF_Exp (float x)
+{
+	return expf (x);
+}
+
+float
+ves_icall_System_MathF_Floor (float x)
+{
+	return floorf (x);
+}
+
+float
+ves_icall_System_MathF_Log (float x)
+{
+	return logf (x);
+}
+
+float
+ves_icall_System_MathF_Log10 (float x)
+{
+	return log10f (x);
+}
+
+float
+ves_icall_System_MathF_Pow (float x, float y)
+{
+	return powf (x, y);
+}
+
+float
+ves_icall_System_MathF_Sin (float x)
+{
+	return sinf (x);
+}
+
+float
+ves_icall_System_MathF_Sinh (float x)
+{
+	return sinh (x);
+}
+
+float
+ves_icall_System_MathF_Sqrt (float x)
+{
+	return sqrtf (x);
+}
+
+float
+ves_icall_System_MathF_Tan (float x)
+{
+	return tanf (x);
+}
+
+float
+ves_icall_System_MathF_Tanh (float x)
+{
+	return tanh (x);
+}
+
+float
+ves_icall_System_MathF_FMod (float x, float y)
+{
+	return fmodf (x, y);
+}
+
+float
+ves_icall_System_MathF_ModF (float x, float *d)
```

Comment:
This icall must null check `d`

Reply:
I changed this to be ref, do we still need the check in that case?

CoreCLR does not have that.

Reply:
If managed was using a byref instead of a pointer, we won't need it as we emit caller side checks. That doesn't happen with pointers.

Reply:
Looks like there's it's only used by Truncate and Round with locals, so it's just lousy code on their end but we should be safe.

Reply:
Added a check, just for sanity's sake.